hydroflow/src/builder/surface/mod.rs

```diff
@@ -44,10 +44,17 @@ use std::hash::Hash;
 
 use crate::lang::lattice::{LatticeRepr, Merge};
 use crate::scheduled::context::Context;
+use crate::scheduled::graph::DirectedEdgeSet;
 use crate::scheduled::handoff::handoff_list::{PortList, PortListSplit};
 use crate::scheduled::port::{RECV, SEND};
 use crate::scheduled::type_list::Extend;
 
+pub trait TrackPushDependencies {
+    fn insert_dep(&self, e: &mut DirectedEdgeSet) -> u16; // id of the inserted node
+}
+pub trait TrackPullDependencies {
```

Comment:
nit: this seems more to me about reifying the topology of the graph rather than "tracking dependencies," maybe a name like `RecordPullTopology` or something? Not sure. Not super strong feels but worth a second thought.

Reply:
Changed to `KeepDataflowGraph`.
Changed `DirectedEdgeSet` to `DataflowGraphStorage`

Reply:
Changed again to `AssembleFlowGraph` and `FlowGraph`